pkg/controllers/csi/driver/volumes/app/publisher.go

```diff
@@ -86,63 +86,63 @@ func (publisher *AppVolumePublisher) PublishVolume(ctx context.Context, volumeCf
 }
 
 func (publisher *AppVolumePublisher) UnpublishVolume(ctx context.Context, volumeInfo *csivolumes.VolumeInfo) (*csi.NodeUnpublishVolumeResponse, error) {
-	volume, err := publisher.loadVolume(ctx, volumeInfo.VolumeID)
+	osMount, err := publisher.db.ReadOSMount(ctx, metadata.OSMount{VolumeMeta: metadata.VolumeMeta{ID: volumeInfo.VolumeID}})
```

Comment:
Then, I guess we're dealing with `VolumeMeta` here instead?

Reply:
[AppMount](https://github.com/Dynatrace/dynatrace-operator/blob/b26c4762d2009e7f945f8f8093ace12588d9a64e/pkg/controllers/csi/metadata/models.go#L53-L61)

Reply:
:+1: I got confused because the 'old' `loadVolume` function reads a 'volume' record from the DB. And now, with the DB migration file, we create one `VolumeMeta` and one `OSMount` per `volume` entry.

Reply:
You are totally right, my mistake :+1:

Reply:
I consider the old/current database logic as [moon logic](https://www.haggardhawks.com/post/moon-logic), so I am not surprised that most people get super confused by it 😅

Reply:
Fixed